### ISSUE TITLE
[Android] [discuss] Change timepicker in Business hours editor to spinners

### ISSUE THREAD

**Issue by strump** [organicmaps/organicmaps#2482]
**Steps to reproduce:**
1. Select POI
2. Click "Edit"
3. Click "Edit business hours"
4. Click on "Open" or "Closed" times.

I don't like round timepicker in Android:
<img src="https://user-images.githubusercontent.com/720808/165925501-53a8379b-8325-442c-b66b-0a9c4f827afc.jpg" width="300px"/>

I prefer spinner:
<img src="https://user-images.githubusercontent.com/720808/165925550-921d5231-1133-4f0b-938d-c3c24d47e020.jpg" width="300px"/>
This would be a one line change in `fragment_timetable_picker.xml`:

```xml
 <TimePicker
    android:id="@+id/picker"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:background="?android:windowBackground"
    android:timePickerMode="spinner"/> <!-- Add this attribute -->
```

Maybe this behaviour should be configurable through app settings?

**Comment by dbf256** [comment]
I use time editor quite often. I don't know what do you mean by "I don't like round timepicker in Android" - design issue or something else, but it is very convenient to set time with just one click to a point in screen instead of scrolling of a spinner.

**Comment by biodranik** [comment]
So looks like we need an alternative :) Let's make it switchable then, but not from the app's settings, but from this screen itself. There are no time pickers in other dialogs, right? This is the only one.

**Comment by strump** [comment]
@dbf256 for me it's hard to pick right number. Especially when I try to tap small numbers in inner surcle (from 13 to 00). Also if I selected wrong hour then timepicker switches to minutes automatically and I have to click on hours to get back.

I think letting user choose Circle or Spinner would solve my issues.

**Comment by dbf256** [comment]
> So looks like we need an alternative :) Let's make it switchable then, but not from the app's settings, but from this screen itself. There are no time pickers in other dialogs, right? This is the only one.

Yes, I don't know other timepickers. There is already a selector in this screen (keyboard at bottom-left), so maybe we can add another one option without complication of the UI.

**Comment by Markus40** [comment]
> @dbf256 for me it's hard to pick right number. Especially when I try to tap small numbers in inner surcle (from 13 to 00). Also if I selected wrong hour then timepicker switches to minutes automatically and I have to click on hours to get back.
> 
> I think letting user choose Circle or Spinner would solve my issues.

You don't have to tap but can hold your finger on the clock, move it to the right number and release. This will let you always select the right time and is a lot faster than the spinner.
